backend/app.py

```diff
@@ -171,49 +170,167 @@ def jqQuery(objtype: str, query: str, objlist=None, ) -> Union[Dict, List]:
-#     filterFields = validFieldList(objlist=objlist, fields=valFilters.keys())
-#     for filterField in filterFields:
-#         filtVals = valFilters[filterField]
-#         if len(filtVals):
-#             condition = 'or'.join([f' .codeset_id == {val} ' for val in filtVals])
-#             clause = f'select({condition})'
-#             queryClauses.insert(0, clause)
-#
-#     queryClauses.insert(0, '.[]')
-#     query = ' | '.join(queryClauses)
-#     subset = jqQuery(objtype=objtype, objlist=objlist, query=query)
-#
-#     return subset
-#     # groupQuery = 'group_by(.concept_set_name) | map({ key: .[0].concept_set_name | tostring, value: [.[] | {version, codeset_id}] }) | from_entries'
-#     # res = jqQuery(objtype=objtype, objlist=subset, query=groupQuery)
-#     # return res
+# TODO: Finish this route
+# TODO: i. Fix: Very slow on large N. For one attempt, I got back ~7,000 results for 'concepts' variables. I think this is
+#  because one of the csets was 'immunotherapy'; a big set.
+# TODO: ii. Color table
+# Example: http://127.0.0.1:8000/concept-set-overlap-table-data?codeset_id=314083061|728628308|98365468
+@APP.get("/concept-set-overlap-table-data")
+def concept_overlap_table_data(
+    codeset_id: Union[str, None] = Query(default=[]),
+) -> List[Dict]:
+    """Concept overlap table: hierarchical"""
+    # TODO: copy/paste over code from /concept-set-overlap-table-data-simple-hierarchy
+    #  - then, repurpose to use `concept_relationship_is_a.csv`
 
+    return []
 
-# TODO: New route
+
+# todo: (i) and (ii) in route '/concept-set-overlap-table-data' apply to this as well
+# Example: http://127.0.0.1:8000/concept-set-overlap-table-data-simple-hierarchy?codeset_id=314083061|728628308|98365468
+@APP.get("/concept-set-overlap-table-data-simple-hierarchy")
+def concept_overlap_table_data_simple_hierarchy(
+    codeset_id: Union[str, None] = Query(default=[]),
+) -> List[Dict]:
+    """Concept overlap table: hierarchical"""
+    # Load data
+    # indent_string = '->'  # looked weird
+    # indent_string = '   '  # @Siggie: As I feared, this gets trimmed / reduced to a single space
+    indent_string = '---'
+    concepts_and_concept_sets: Dict = concept_sets_by_concept(codeset_id)
+    concept_set_ids = codeset_id.split('|')
+    concept_set_ids = [int(x) for x in concept_set_ids]
+    df_concept_set_members = DS['concept_set_members']
+    df_concept_ancestor = DS['concept_ancestor']
+    df_concept_set_members_i = df_concept_set_members[df_concept_set_members['codeset_id'].isin(concept_set_ids)]
+    df_concept_ancestor_i = df_concept_ancestor[
+        (df_concept_ancestor.ancestor_concept_id.isin(df_concept_set_members_i.concept_id)) &
+        (df_concept_ancestor.descendant_concept_id.isin(df_concept_set_members_i.concept_id))]
+
+    # Transform: get essential information
+    # todo: account for max_levels_of_separation, or just get direct parent/child relationships
```

Comment:
Only using `min_levels_of_separation` currently. But if the relationship `is_a` hierarchy is better, we'll just use it instead and won't have to worry about this.
> `todo: account for max_levels_of_separation, or just get direct parent/child relationships`